instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/utils.py

```diff
@@ -132,6 +132,11 @@ def attach_span(task, task_id, span, is_publish=False):
     NOTE: We cannot test for this well yet, because we do not run a celery worker,
     and cannot run `task.apply_async()`
     """
+    if task is None:
```

Comment:
I'll copy the comment I wrote on the original PR in your fork for the wide discussion:

> I think the effect of this return statement is that the span (along with activation) will not be recorded into the CTX_KEY, which means that a later invocation of the _trace_after_publish signal will not be able to retrieve_span thus will not end the span.
Am I missing something?

Didn't actually try to run it, so I might just not fully understand the details of what causes the "publish" span to be ended.